independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java

```diff
@@ -309,21 +297,24 @@ public ResultHandle apply(List<InterceptorInfo> interceptors) {
             InterceptionInfo interception = bean.getInterceptedMethods().get(method);
             DecorationInfo decoration = bean.getDecoratedMethods().get(method);
             MethodDescriptor forwardDescriptor = forwardingMethods.get(methodDescriptor);
+            List<Type> parameters = method.parameters();
 
-            if (interception != null) {
-                String methodId = "m" + methodIdx++;
-                ResultHandle methodIdHandle = constructor.load(methodId);
+            // Each method has a corresponding InterceptedMethodMetadata field
+            FieldCreator metadataField = subclass
+                    .getFieldCreator("arc$" + methodIdx++, InterceptedMethodMetadata.class.getName())
```

Comment:
So this field exists even for methods that are not intercepted, just decorated?

Reply:
What do you mean? The field isn't actually written if you get `interception == null`.

Reply:
It's not written, but it exists, and I wonder if it has to.

Reply:
Hmm, yea, technically you could move this declaration inside the `if` condition if that's what you mean.

Reply:
That's what I meant -- just not sure if there's some reason I'm not aware of :-)

Reply:
That's a good point. Of course, we don't need the field for decorated methods...